modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/runtime/connectivity/oauth/OAuthHandlerTestCase.java

```diff
@@ -0,0 +1,158 @@
+  private static final String OTHER_KEY = "1874947571-1840879217-123123123-1745289126";
+
+  private static final List<String> OBJECT_STORE_ENTRIES = Arrays.asList(SOME_KEY, OTHER_KEY);
+  private MuleContextWithRegistry muleContext;
+
+  @InjectMocks
+  AuthorizationCodeOAuthHandler oauthHandler = new AuthorizationCodeOAuthHandler();
+
+  @Mock
+  private ObjectStoreManager storeManager;
+
+  @Mock
+  private ObjectStore<MetadataCache> objectStore;
+
+  private SimpleUnitTestSupportSchedulerService schedulerService;
+  private ConfigurationProperties configurationProperties;
+  private static final String CLIENT_ID = "client_id";
+  private static final String SECRET_ID = "secret_id";
+  private static final String ORG_ID = "org_id";
+  private static final String SERVICE_URL = "service_url";
+  private static final String PLATFORM_AUTH_URL = "http://localhost/accounts";
+
+  @Test
+  @Issue("W-11410770")
+  @Description("Verify that ObjectStore is always created, if none exists a new one is created")
+  public void verifyThatObjectStoreisAlwaysCreated() throws InitialisationException, ObjectStoreException {
+    PlatformManagedOAuthConfig config = PlatformManagedOAuthConfig.from("", "", null, null, null, null, configurationProperties);
+
+    oauthHandler.initialise();
+
+    ObjectStore objectStore = oauthHandler.getObjectStoreLocator().apply(config);
+    assertThat(objectStore, is(notNullValue()));
+    assertThat(objectStore.allKeys(), is(OBJECT_STORE_ENTRIES));
+    verify(storeManager, times(1)).getOrCreateObjectStore(anyString(), any(ObjectStoreSettings.class));
+  }
+
+  @Rule
+  public TemporaryFolder tempWorkDir = new TemporaryFolder();
+
+  @Before
+  public void setup() throws ObjectStoreException {
+    schedulerService = new SimpleUnitTestSupportSchedulerService();
+    muleContext = mock(MuleContextWithRegistry.class);
+
+    when(storeManager.getOrCreateObjectStore(anyString(), any()))
+        .thenReturn(objectStore);
+    when(objectStore.allKeys()).thenReturn(Arrays.asList(SOME_KEY, OTHER_KEY));
+
+    configurationProperties = mock(ConfigurationProperties.class);
+    when(configurationProperties.resolveStringProperty(OCS_CLIENT_SECRET)).thenReturn(of(SECRET_ID));
+    when(configurationProperties.resolveStringProperty(OCS_CLIENT_ID)).thenReturn(of(CLIENT_ID));
+    when(configurationProperties.resolveStringProperty(OCS_ORG_ID)).thenReturn(of(ORG_ID));
+    when(configurationProperties.resolveStringProperty(OCS_SERVICE_URL)).thenReturn(of(SERVICE_URL));
+    when(configurationProperties.resolveStringProperty(OCS_PLATFORM_AUTH_URL)).thenReturn(of(PLATFORM_AUTH_URL));
+    when(configurationProperties.resolveStringProperty(OCS_API_VERSION)).thenReturn(empty());
+    when(configurationProperties.resolveStringProperty(OCS_API_VERSION)).thenReturn(empty());
+  }
+
+
+  @Override
+  protected void doTearDown() throws Exception {
+    oauthHandler.stop();
+    currentMuleContext.set(null);
```

Comment:
What is this needed for?

Reply:
Shouldn't matter much for tests, but we have consistently done this in many tests

Reply:
Leave the test as minimal as needed. It is usually needed in other tests.